### PR TITLE
Slimernalia2019

### PR DESCRIPTION
**Pull request by MeatyLearner**
Just like last year's Slimernalia:

- Food is free

- Roles have been reversed: only juveniles can kill

- You can gamble from anywhere, however...

New this Slimernalia:

- You can buy a Gellphone from the Bazaar, which allows you to gamble from anywhere while you're using it.

- You can now gamble slime! Just add "slime" to the end of the commands you would usually do.

- You can buy wrapping paper at Glocksbury Comics, which you can use to !wrap gifts for other players. They can then !unwrap your gift.

- You may now have slimeoid battles anywhere AND bet slime on them.

- Killing players gives you their sigillaria, a furniture item, instead of their scalp.

- A new cosmetic item, the pileus, has been added to the Bazaar.

- The Slime Casino has been acquired by SlimeCorp, and has been rebranded the SlimeCorp Casino. SlimeCorp takes 20% of all your winnings.

The Kingpin of Slimernalia

- A new leaderboard has been added: Most Festive. Whoever is at the top of this leaderboard becomes the Kingpin of Slimernalia for that day.

- Use !festivity to check your current festivity.

Your festivity increases by doing the following:

- Killing players (1 festivity per slime splattered on districts)

- Wrapping, unwrapping and giving gifts (100 festivity for wrapping and unwrapping, 1000 for giving)

- Gambling (1 festivity per slime gambled, 1 festivity per 1,000,000,000,000 SlimeCoin gambled)

- Eating food (1 festivity per point of hunger restored)

- Owning sigillaria (100 festivity for each sigillaria in your inventory)

**!yoslimernalia**

### CODE REVIEW

**Comment by MeatyLearner** [comment]
for k1p's eyes only:
-fixed a bug where renameable items could be used instead of the expected item (ie: !makecostume costumekit to make recursive costumekit costumes)
-garrote works again
-fixed bugs regarding permissions from the subzone removal update
-knocking triggers a pvp flag of 10 seconds

File: ewcasino.py [hunk 1]
```diff
@@ -2043,7 +2256,7 @@ def determine_trick_taker(trick,gametype,trump):
 				item_select = item_object
 				break
 
-	if usermodel.poi != ewcfg.poi_id_thecasino:
+	if cmd.message.channel.name != ewcfg.channel_realestateagency:
```
Comment (magicksRampage) [inline ewcasino.py]: should be channel_casino

File: ewcmd.py [hunk 2]
```diff
@@ -1335,12 +1335,154 @@ def item_off(id_item, id_server, item_name = "", is_pushed_off = False):
...
+		return await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, response))
+	
+	gift_message = cmd.tokens[2]
+	
+	item_search = ewutils.flattenTokenListToString(cmd.tokens[3:])
+	item_sought = ewitem.find_item(item_search=item_search, id_user=cmd.message.author.id, id_server=cmd.message.server.id)
+
+	if item_sought:
+		item = ewitem.EwItem(id_item=item_sought.get('id_item'))
+		if item.item_type == ewcfg.it_furniture:
```
Comment (magicksRampage) [inline ewcmd.py]: gifts are created as general item types, not furniture

File: ewitem.py [hunk 3]
```diff
@@ -1088,6 +1088,19 @@ def inventory(
...
+			elif context == ewcfg.item_id_gellphone:
+				try:
+					gellphone_role = ewutils.return_server_role(server, ewcfg.role_gellphone_proper)
+					
+					if ewutils.check_user_has_role(server, author, ewcfg.role_gellphone_proper):
+						response = "You turn off your gellphone."
+						await cmd.client.remove_roles(author, gellphone_role)
+					else:
+						response = "You turn on your gellphone."
+						await cmd.client.add_roles(author, gellphone_role)
```
Comment (magicksRampage) [inline ewitem.py]: I really dislike this way of dealing with the gellphone role. we have all role changes unified in ewrolemgr.updateRoles() for a good reason. this kind of thing is a pain to debug and maintain

Comment (magicksRampage) [reply ewitem.py]: instead, give gellphones an item_prop that says whether it's on or off and check for that in updateRoles()

File: ewmap.py [hunk 4]
```diff
@@ -1249,11 +1249,12 @@ def retrieve_locked_districts(id_server):
 	Get information about an adjacent zone.
 """
 async def scout(cmd):
-	if channel_name_is_poi(cmd.message.channel.name) is False or fetch_poi_if_coordless(cmd.message.channel.name) is not None:
+	user_data = EwUser(member=cmd.message.author)
+	user_poi = ewcfg.id_to_poi.get(user_data.poi)
+	
+	if channel_name_is_poi(cmd.message.channel.name) is False or user_poi.is_subzone:
```
Comment (magicksRampage) [inline ewmap.py]: user_poi is never a subzone. it's always the main district

**Review by magicksRampage — CHANGES_REQUESTED** [review]

File: ewitem.py (hunk 3 above)
Comment (magicksRampage) [reply ewitem.py]: the same goes for the slimernalia kingpin role actually

**Pull request merged**